packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -333,6 +333,11 @@ class RunCommand extends RunCommandBase {
       throwToolExit('Error: --train is only allowed when running as --dynamic --profile '
           '(recommended) or --debug (may include unwanted debug symbols).');
 
+    List<String> expFlags = null;
+    if (argParser.options.containsKey(FlutterOptions.kEnableExperiment) &&
+        argResults[FlutterOptions.kEnableExperiment] != null) {
+      expFlags = argResults[FlutterOptions.kEnableExperiment];
```

Comment:
enable-experiment being MultiOption won't ever have `null` value. it is always a list, it can be 0-length list if no values are specified for this option